src/writeckpt.cpp

```diff
@@ -144,6 +144,20 @@ mtcp_writememoryareas(int fd)
     // never get back to this function and the object is never released.
     delete procSelfMaps;
   }
+  /* The restoreBuf memory region might have merged with neighbor memory regions
+   * with same permissions (PROT_NONE). We will unmap this region before
+   * processing all /proc/self/maps regions. And, will map this region back
+   * afterwards to reserve the same area.
+   * Unmap-map works as there's no data in the region to lose in unmapping.
+   *
+   * Note: We could achieve the same results by adding a guard page at each end.
+   *       However, this is a more cleaner solution.
+   */
+  JASSERT(munmap((void *)ProcessInfo::instance().restoreBufAddr(),
+            ProcessInfo::instance().restoreBufLen()) == 0)
+    ((void *)ProcessInfo::instance().restoreBufAddr())
+    (ProcessInfo::instance().restoreBufLen())
+    (JASSERT_ERRNO);
 
   /* Finally comes the memory contents */
   procSelfMaps = new ProcSelfMaps();
```

Comment:
Thanks for the patch, @JainTwinkle! This is an important fix.

However, I think we need to do a bit more work here. The `new` operator might end up putting `ProcSelfMaps` object in the area that we just unmapped. This could lead to weird issues.

I suggest the following:
1. Move ProcSelfInfo constructor code (https://github.com/dmtcp/dmtcp/blob/master/src/procselfmaps.cpp#L57-L91) to a separate `init()` function.
2. Call the constructor before unmapping the reserved region.
3. Call `init()` after unmapping.

Does that sound reasonable to you?

Reply:
@rohgarg, In the end @JainTwinkle and I decided that her previous solution using guard pages was the simpler solution.  So, we're going back to the "guard pages" around the restoreBuf.